plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/expression/TopN.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch.expression;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.connector.SortOrder;
+import org.elasticsearch.search.sort.SortBuilder;
+import org.elasticsearch.search.sort.SortBuilders;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class TopN
+{
+    public static final long NO_LIMIT = -1;
+    public static final TopN EMPTY = new TopN(NO_LIMIT, new ArrayList<>());
+
+    private final long limit;
+    private final List<TopNSortItem> topNSortItems;
+
+    public static TopN fromLimit(long limit)
+    {
+        return new TopN(limit, new ArrayList<>());
+    }
+
+    @JsonCreator
+    public TopN(@JsonProperty("limit") long limit,
+            @JsonProperty("topNSortItems") List<TopNSortItem> topNSortItems)
+    {
+        this.limit = limit;
+        this.topNSortItems = requireNonNull(topNSortItems, "topNSortItems is null");
+    }
+
+    @JsonProperty("limit")
+    public long getLimit()
+    {
+        return limit;
+    }
+
+    @JsonProperty("topNSortItems")
+    public List<TopNSortItem> getTopNSortItems()
+    {
+        return topNSortItems;
+    }
+
+    public TopN addSortItem(TopNSortItem sortItem)
+    {
+        topNSortItems.add(sortItem);
+        return this;
+    }
```

Comment:
It is always a bit recommended to make the object `TopN` immutable instead of changing its state here